src/molecules/lists/drag-and-drop-list-box.tsx

```diff
@@ -0,0 +1,212 @@
+                <Paragraph>No Items Added</Paragraph>
+            </div>
+        );
+    }
+
+    const processReordered = (
+        items: Array<ListBoxItem<T>>,
+        startIndex: number,
+        endIndex: number
+    ): Array<ListBoxItem<T>> => {
+        const result = [...items];
+        const [removedItem] = result.splice(startIndex, 1);
+        result.splice(endIndex, 0, removedItem);
+
+        return result;
+    };
+
+    const handleDragEnd = (result: DropResult) => {
+        if (
+            result?.destination == null ||
+            result?.source == null ||
+            result.source.index === result.destination.index
+        ) {
+            return;
+        }
+
+        props.onReordered(
+            processReordered(
+                props.items,
+                result.source.index,
+                result.destination!.index
+            )
+        );
+    };
+
+    /**
+     * if we don't memoize the child rendering, then the <Draggable> state
+     * sometimes does not update properly. Without memoization, if you drag
+     * and drop a <Draggable> outside it's <Droppable>, then the <Draggable> will
+     * still think it's being dragged, and apply the -dragging class, even though
+     * it is no longer being dragged. Memoization of rendering the list fixes
+     * this issue, and also increases performance.
+     */
+    const DraggableItemList = React.memo(function(itemProps: {
+        items: Array<ListBoxItem<T>>;
+    }) {
+        const cssIsDragging = (snapshot: DraggableStateSnapshot) =>
+            snapshot.isDragging ? "-dragging" : "";
+
+        return (
+            <React.Fragment>
+                {itemProps.items.map((item: ListBoxItem<T>, index: number) => (
+                    <Draggable
+                        draggableId={`${item.id}`}
+                        index={index}
+                        key={uuid.v4()}>
+                        {(
+                            provided: DraggableProvided,
+                            snapshot: DraggableStateSnapshot
+                        ) => (
+                            <div
+                                {...provided.draggableProps}
+                                {...provided.dragHandleProps}
+                                ref={provided.innerRef}
+                                className={`${ListBoxItemClassName} ${cssIsDragging(
+                                    snapshot
+                                )}`}
+                                style={provided.draggableProps.style}>
+                                <div className="-drag-handle">
+                                    <Icon
+                                        type={Icons.DragAndDrop}
+                                        size={IconSizes.Large}
+                                    />
+                                </div>
+                                {// if
+                                item.label != null && (

```

Comment:
Question - should this leverage `StringUtils.hasValue` instead of a plain null check?